Automated Smoketest
Sha: cb9809891a0802058273274839be3721bf14985c
Build: failed
Number of Error(s): 13
  - \xe2\x80\x98error           - warnings are errors\xe2\x80\x99
  - \xe2\x80\x98error           - warnings are errors\xe2\x80\x99
c++: error: /mnt/disk1/home/vamosax/smoketest/PR-9392/build/ecl/hql/hqlgram.cpp: No such file or directory
c++: fatal error: no input files
make[2]: *** [ecl/hql/CMakeFiles/hql.dir/hqlgram.cpp.o] Error 4
/mnt/disk1/home/vamosax/smoketest/PR-9392/HPCC-Platform/ecl/hql/hqlgram2.cpp:29:21: fatal error: hqlgram.h: No such file or directory
/mnt/disk1/home/vamosax/smoketest/PR-9392/HPCC-Platform/ecl/hql/hqlparse.cpp:27:21: fatal error: hqlgram.h: No such file or directory
/mnt/disk1/home/vamosax/smoketest/PR-9392/HPCC-Platform/ecl/hql/hqllex.l:32:21: fatal error: hqlgram.h: No such file or directory
make[2]: *** [ecl/hql/CMakeFiles/hql.dir/hqlgram2.cpp.o] Error 1
make[2]: *** [ecl/hql/CMakeFiles/hql.dir/hqlparse.cpp.o] Error 1
make[2]: *** [ecl/hql/CMakeFiles/hql.dir/hqllex.cpp.o] Error 1
make[1]: *** [ecl/hql/CMakeFiles/hql.dir/all] Error 2
make: *** [all] Error 2

--- a/cmake_modules/commonSetup.cmake
+++ b/cmake_modules/commonSetup.cmake
@@ -618,6 +618,13 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
         MESSAGE(FATAL_ERROR "You need bison version 2.4.1 or later to build this project (version ${BISON_VERSION} detected)")
       ENDIF()
 
+      IF ("${BISON_VERSION}" VERSION_LESS "2.7.0")
+        #Ignore all warnings - not recommend to develope on this version!
+        SET(bisonopt "-Wnone")
+      ELSE()
+        SET(bisonopt -Werror -Wno-other)
+      ENDIF()
+
       IF ("${FLEX_VERSION}" VERSION_LESS "2.5.35")
         MESSAGE(FATAL_ERROR "You need flex version 2.5.35 or later to build this project (version ${FLEX_VERSION} detected)")
       ENDIF()

--- a/ecl/hql/CMakeLists.txt
+++ b/ecl/hql/CMakeLists.txt
@@ -111,16 +111,16 @@ include_directories (
 
 if (WIN32)
     add_custom_command ( OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/hqlgram.cpp ${CMAKE_CURRENT_BINARY_DIR}/hqlgram.h
-        #pipe result through findstr to remove warnings that are hard to suppress, and pipe through more to prevent error code from no matches aborting the compile
-        COMMAND ${bisoncmdprefix} ${bisoncmd} --report=state --defines=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.h --output=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.cpp ${CMAKE_CURRENT_SOURCE_DIR}/hqlgram.y 2>&1 | findstr /V "unused value" | more
+        #${bisonopt} is used to force s/r and r/r errors to abort compiling
+        COMMAND ${bisoncmdprefix} ${bisoncmd} ${bisonopt} --report=state --defines=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.h --output=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.cpp ${CMAKE_CURRENT_SOURCE_DIR}/hqlgram.y 2>&1
         DEPENDS hqlgram.y
     )
 else()
     add_custom_command ( OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/hqlgram.cpp ${CMAKE_CURRENT_BINARY_DIR}/hqlgram.h
         #delete output files before building to force rebuilds with errors to fail rather than continue unnoticed, using the existing output files.
         COMMAND rm -f ${CMAKE_CURRENT_BINARY_DIR}/hqlgram.h ${CMAKE_CURRENT_BINARY_DIR}/hqlgram.cpp
-        #pipe result through grep to remove warnings that are hard to suppress, and pipe through cat to prevent error code from no matches aborting the compile
-        COMMAND ${bisoncmdprefix} ${bisoncmd} --report=state --defines=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.h --output=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.cpp ${CMAKE_CURRENT_SOURCE_DIR}/hqlgram.y 2>&1 | grep -v "unused value" | cat
+        #${bisonopt} is used to force s/r and r/r errors to abort compiling
+        COMMAND ${bisoncmdprefix} ${bisoncmd} ${bisonopt} --report=state --defines=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.h --output=${CMAKE_CURRENT_BINARY_DIR}/hqlgram.cpp ${CMAKE_CURRENT_SOURCE_DIR}/hqlgram.y 2>&1
         DEPENDS hqlgram.y
     )
 endif()

--- a/ecl/hql/hqlgram.y
+++ b/ecl/hql/hqlgram.y
@@ -4093,7 +4093,7 @@ funcRetType
                             $$.setType(makeRowType(expr->getType()));
                             $$.setPosition($1);
                         }
-    | typeDef
+    | userTypedefType
     ;
 
 payloadPart

--- a/ecl/regress/issue16626.ecl
+++ b/ecl/regress/issue16626.ecl
@@ -1,0 +1,29 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2013 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+array_t := set of real8;
+double := real8;
+
+x := SERVICE : library('supermaths')
+   array_t transpose(array_t value, unsigned cols, unsigned rows);
+   double square(double x);
+END;
+
+
+output(x.transpose([1,2,3,4],2,2));
+output(x.square(10.0));

